src/lib/schemas.ts

```diff
@@ -1,26 +1,71 @@
 import { z } from 'zod';
+import { activeContractsList } from '@aragon/osx-ethers';
 
 // Define the schema for the private key
 export const privateKeySchema = z
   .string()
   .length(64, 'Private key must be exactly 64 characters long.')
-  .regex(
-    /^[a-fA-F0-9]*$/,
-    'Private key must only contain hexadecimal characters.',
-  );
+  .regex(/^[a-fA-F0-9]*$/, 'Private key must only contain hexadecimal characters.');
 
 // Define the schema for the Tenderly key
 export const tenderlyKeySchema = z
   .string()
   .length(32, 'Tenderly key must be exactly 32 characters long.')
-  .regex(
-    /^[a-zA-Z0-9-]*$/,
-    'Tenderly key must only contain numbers and letters.',
-  );
+  .regex(/^[a-zA-Z0-9-]*$/, 'Tenderly key must only contain numbers and letters.');
 
 export const contractNameSchema = z
   .string()
   .nonempty({ message: 'Contract name cannot be empty.' })
   .regex(/^[a-zA-Z0-9]*$/, {
     message: 'Contract name can only contain numbers and letters.',
   });
+
+export const ethereumAddressSchema = z.string().refine(
```

Comment:
lets add some typing here.

Reply:
and to all variables below

Reply:
honestly I think its best to leave it to zod here. One of the main benefits of using zod here other that the error handling is for the type infrence. For these the type is easy, but some of the others are massive. 

we dont loose any of the typescript magic by not explisitly using the type here